### PR TITLE
[linux_proc_extras] backport linux_proc_extras changes from the sdk

### PR DESCRIPTION
**Pull request by gmmeyer**
I made a few changes to linux_proc_extras when I moved it to the sdk. It makes it easier to test. Since this is still the source of truth, we have to backport it here.

### CODE REVIEW

File: checks.d/linux_proc_extras.py [hunk 1]
```diff
@@ -24,47 +25,66 @@
 }
 
 class MoreUnixCheck(AgentCheck):
+    tags = []
+    proc_path_map = {
```
Comment (degemer) [inline checks.d/linux_proc_extras.py]: Is it used?

File: checks.d/linux_proc_extras.py [hunk 2]
```diff
@@ -24,47 +25,66 @@
 }
 
 class MoreUnixCheck(AgentCheck):
+    tags = []
```
Comment (degemer) [inline checks.d/linux_proc_extras.py]: Same, it's set anyway at every check, unless I'm missing something?

File: checks.d/linux_proc_extras.py [hunk 3]
```diff
@@ -24,47 +25,66 @@
...
+        with open(self.proc_path_map['stat_info'], 'r') as stat_info:
             lines = [line.strip() for line in stat_info.readlines()]
-
             for line in lines:
                 if line.startswith('ctxt'):
                     ctxt_count = float(line.split(' ')[1])
-                    self.monotonic_count('system.linux.context_switches', ctxt_count, tags=tags)
+                    self.monotonic_count('system.linux.context_switches', ctxt_count, tags=self.tags)
                 elif line.startswith('processes'):
+                    self.log.info(line)
```
Comment (degemer) [inline checks.d/linux_proc_extras.py]: Do we really want this at INFO level? Side note: if we're only reading the log, without knowing the check, this line doesn't make sense. Could we prefix it with what it is/means?

File: checks.d/linux_proc_extras.py [hunk 4]
```diff
@@ -24,47 +25,66 @@
...
                     process_count = int(line.split(' ')[1])
-                    self.monotonic_count('system.linux.processes_created', process_count, tags=tags)
+                    self.log.info(process_count)
```
Comment (degemer) [inline checks.d/linux_proc_extras.py]: Same

File: checks.d/linux_proc_extras.py (hunk 1 above)
Comment (gmmeyer) [inline checks.d/linux_proc_extras.py]: It's used in a bunch of places.

File: checks.d/linux_proc_extras.py (hunk 2 above)
Comment (gmmeyer) [inline checks.d/linux_proc_extras.py]: This I can get rid of

File: checks.d/linux_proc_extras.py (hunk 3 above)
Comment (gmmeyer) [inline checks.d/linux_proc_extras.py]: Yup I'll remove it thanks!

File: checks.d/linux_proc_extras.py (hunk 4 above)
Comment (gmmeyer) [inline checks.d/linux_proc_extras.py]: Yup I'll remove it thanks!

File: checks.d/linux_proc_extras.py (hunk 1 above)
Comment (gmmeyer) [inline checks.d/linux_proc_extras.py]: It's also static, the tags thing was wrong as it's not static, but this makes sense to include here

Comment (gmmeyer) [inline checks.d/linux_proc_extras.py]: Actually it's redefined, so no I don't think it is. You're right.

**Pull request merged**